pennylane/transforms/metric_tensor.py

```diff
@@ -65,6 +71,8 @@ def metric_tensor(tape, approx="block-diag", diag_approx=None, allow_nonunitary=
             This keyword argument is deprecated in favor of ``approx`` and will be removed soon
         allow_nonunitary (bool): Whether non-unitary operations are allowed in circuits
             created by the transform. Only relevant if ``approx`` is ``None``
+        aux_wire (int or pennylane.wires.Wires): Auxiliary wire to be used for
+            Hadamard tests. Defaults to ``tape.num_wires`` if not provided

```

Comment:
I see. The suggestion would be a bit confusing though, because the aux wire is _not_ part of the original circuit.
And the last wire of the device might differ from `tape.num_wires` if the device is larger than minimally required. Also, if custom wire tags are used, "last" wire is not unambiguous :see_no_evil: 
I'll think of a formulation that does not rely on tapes underlying the construction :)